app/code/Magento/CatalogUrlRewrite/Observer/AfterImportDataObserver.php

```diff
@@ -475,7 +475,7 @@ private function currentUrlRewritesRegenerate()
             ]
         );
 
-        $urlRewrites = [];
+        $urlRewriteArrays = [];
```

Comment:
@hostep, maybe we should use mergeDataProvider, similar to the following code? What do you think?
https://github.com/magento/magento2/blob/7c6b6365a3c099509d6f6e6c306cb1821910aab0/app/code/Magento/UrlRewrite/Model/CompositeUrlFinder.php#L74-L83

Reply:
That could work indeed, I'll try it out a bit later